blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/InvalidatingJobCache.java

```diff
@@ -0,0 +1,44 @@
+package io.jenkins.blueocean.rest.impl.pipeline;
+
+import com.google.common.cache.Cache;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.listeners.ItemListener;
+import io.jenkins.blueocean.commons.ServiceException;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+@Restricted(NoExternalUse.class)
+@Extension
+public class InvalidatingJobCache<T> extends ItemListener {
```

Comment:
Is there way to verify/test if a PR changes and is notified thru webohook `ItemListener` gets invoked? Hopefully it does, otherwise we might be in stale state up to an hour. Same for reindexing.